text/0000-derive-copy-and-clone.md

```diff
@@ -0,0 +1,44 @@
+
+In the case of `Eq` and `Ord`, the motivation is different. Implementors of either of these traits must first implement their partial versions, which means that no user can accidentally omit them. However, it is our subjective experience that no user who derives `Eq` will have manually implemented `PartialEq`; likewise for `Ord` and `PartialOrd`. Therefore this aspect of this RFC is solely to improve ergonomics, and should be considered secondary in importance to the issue of `Copy` and `Clone`.
+
+# Detailed design
+
+The `derive` syntax extension must be modified to accomodate these hardcoded rules. The compiler must be able to accept both `#[derive(Copy, Clone)]` and `#[derive(Copy)]`, the latter of which will implicitly insert a derived implementation of `Clone`. Like wise the compiler must accept both `#[derive(Eq)]` and `#[derive(PartialEq, Eq)]`, as well as both `#[derive(Ord)]` and `#[derive(PartialOrd)]`. Despite not appearing in the source, the automatically inserted implementations must be visible to rustdoc.
+
+An implementation of this RFC can be seen here: https://github.com/rust-lang/rust/pull/23905
+
+# Drawbacks
+
+Makes certain derived traits "special", forming a pseudo-inheritance hierarchy of sorts.
+
+Rustdoc will show implementations of traits with no direct origin in the source.
+
+# Alternatives
+
+The most straightforward alternative would be to accept only the `Copy`/`Clone` aspect of this RFC, as it is concerned with library correctness (in the "const correctness" sense) whereas the `Ord`/`PartialOrd` and `Eq`/`PartialEq` cases are merely for ergonomics.
+
+If the above alternative is desired, then it may make sense, for consistency's sake, to cause the compiler to return an error when it encounters `#[derive(Copy)]`, with a message demanding that the user modify the attribute to `#[derive(Copy, Clone)]`. This would address the drawback of there being an unclear origin for the `Clone` trait when viewed from rustdoc. It would also have symmetry with the demand for `#[derive(PartialOrd, Ord)]`, again assuming that the alternative from the prior paragraph is desired.
+
+# Unresolved questions
+
+Is it reasonable to assume that no user who automatically derives `Eq`/`Ord` has manually implemented `PartialEq`/`PartialOrd`? If this assumption fails to hold, what behavior should result?
```

Comment:
Cargo has a [few examples](https://github.com/rust-lang/cargo/blob/84d6d2c7d6d8c28e3c1b10c2ec43032f75fd3c44/src/cargo/core/source.rs#L66-L78) of deriving `Eq` with a manual implementation of `PartialEq`. I'd be fine just moving some words from here to the "Drawbacks" section, though!